src/pages/workspaces/List.js

```diff
@@ -213,12 +263,31 @@ export const WorkspaceList = ajaxCaller(class WorkspaceList extends Component {
               onClick: () => this.setState({ creatingNewWorkspace: true })
             }),
             _.map(workspace => {
-              return h(WorkspaceCard, { listView, workspace, key: workspace.workspace.workspaceId })
+              return h(WorkspaceCard, {
+                listView,
+                onClone: () => this.setState({ cloningWorkspaceName: workspace }),
+                onDelete: () => this.setState({ deletingWorkspaceName: workspace }),
+                onShare: () => this.setState({ sharingWorkspaceName: workspace }),
```

Comment:
Two points here: First, as written, these variables seem to have confusing names: I would expect them to contain e.g. the *name* of the workspace being shared, but instead they contain a reference to the whole workspace object.

Second, stepping back a bit, we have an opportunity here to use less state: we don't actually need to store the whole workspace object. If we just store a unique ID, then we can use that to look up the full object later, as needed. This means a bit more work at render time, but a simpler state model because there's no duplication of data. That feels like a win to me.